### PR TITLE
Adds a folder for the 2023 ECP Tutorial and adds material for DYAD

### PR DESCRIPTION
**Pull request by ilumsden**
This PR adds a new folder for the 2023 ECP Tutorial of Flux. This folder contains the same material used in the RADIUSS demo, plus a few extra files for DYAD. More specifically, this PR makes the following changes to the material:

* Adds code to `Dockerfile.spawn` to clone and build DYAD and copy DYAD's demo applications
* Adds code to `Dockerfile.spawn` to build DYAD's demo applications
* Adds a Jupyter Notebook (called `dyad-aws.ipynb`) containing the material for the DYAD demo
* Adds a couple of `.svg` files to be displayed within `dyad-aws.ipynb`

The rest of the material in the new `2023-ECP` directory is the same as the material from the RADIUSS demo.

### CODE REVIEW

File: 2023-ECP/JupyterNotebook/README.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
+To run the AWS tutorial, visit https://tutorial.flux-framework.org 
+You can use any login you want, but choose something relatvely uncommon 
+(like your email address) or you may end up sharing a JupyterLab 
+instance with another user. The tutorial password will be provided to you. 
+
+If you have Docker available, you can build and run the tutorial with:
+
+```
+docker build --build-arg GLOBAL_PASSWORD=<password of your choice> -t <desired image name> -f Dockerfile.hub .
+docker build -t <desired image name> -f Dockerfile.spawn .
```
Comment (JaeseungYeom) [inline 2023-ECP/JupyterNotebook/README.md:10]: It should clear that <desired image name> here is different from the one above. They are for two different images and thus two different names. We can change it to for example, `<desired_image_name_for_spawner>`

File: 2023-ECP/JupyterNotebook/dyad-aws.ipynb [hunk 2]
```diff
@@ -0,0 +1,742 @@
...
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "81d7d87f-1e09-42c8-b165-8902551f6847",
+   "metadata": {},
+   "source": [
+    "# This concludes the notebook tutorial for DYAD.\n",
+    "\n",
+    "## If you are interested in learning more about DYAD, check out our [ReadTheDocs page](https://dyad.readthedocs.io/en/latest/) or [our short paper](https://dyad.readthedocs.io/en/latest/_downloads/27090817b034a89b76e5538e148fea9e/ShortPaper_2022_eScience_LLNL.pdf) and [poster](https://dyad.readthedocs.io/en/latest/_downloads/1f11761622683662c33fe0086d1d7ad2/Poster_2022_eScience_LLNL.pdf) from eScience 2022."
```
Comment (JaeseungYeom) [inline 2023-ECP/JupyterNotebook/dyad-aws.ipynb:709]: Do we have link to the actual code repository?

**Comment by milroy** [comment]
@ilumsden : I'll review once I test tomorrow, but you'll need to delete `2023-ECP/Flux_RADIUSS_tutorial_2022.pdf`.

File: 2023-ECP/JupyterNotebook/radiuss-aws-flux.ipynb [hunk 3]
```diff
@@ -0,0 +1,1366 @@
+{
```
Comment (milroy) [inline 2023-ECP/JupyterNotebook/radiuss-aws-flux.ipynb:1]: Please change the filename to ecp23-flux.ipynb.

File: 2023-ECP/JupyterNotebook/dyad-aws.ipynb [hunk 4]
```diff
@@ -0,0 +1,742 @@
+{
```
Comment (milroy) [inline 2023-ECP/JupyterNotebook/dyad-aws.ipynb:1]: A better name for the file might be `ecp23-flux-dyad.ipynb`.

**Review by milroy — CHANGES_REQUESTED** [review]
Repeating in the review: please delete 2023-ECP/Flux_RADIUSS_tutorial_2022.pdf. Apart from renaming two files this looks good.

**Comment by ilumsden** [comment]
@JaeseungYeom @milroy I've made the requested changes.

Also, @JaeseungYeom, I gave a dry run of the DYAD demo in the GCLab group meeting today, and someone recommended some changes to `dyad_example1.svg`. In particular, they recommended:

* Change "P1" and "P2" to something like "Node 1" and "Node 2"
* Change "Task 1", "Task 2", and "Task 3" to use the terms "Producer" and "Consumer"
* Change the background to be clear (the background is white, so it looks a bit odd if you have dark mode enabled in Jupyter)

Not sure if we want to worry about that at this point, but I thought I'd bring it up just in case.

**Comment by ilumsden** [comment]
I just pushed the figure change stuff.

**Review by milroy — APPROVED** [review]
LGTM! Merging.

**Pull request merged**